tools/static-nodes.c

```diff
@@ -252,8 +252,10 @@ static int do_static_nodes(int argc, char *argv[])
 		if (buf[0] == '#')
 			continue;
 
+		char format[22];
+		sprintf(format, "%%%ds %%%ds %%c%%u:%%u", PATH_MAX - 1, PATH_MAX - 1);
```

Comment:
If this commit is about protecting us against a `PATH_MAX` which is not 4096, then this `sprintf` call is actually dangerous. You could overflow the `format` buffer if PATH_MAX is sufficiently large.

Yet it's also just theoretical. A system with such a huge `PATH_MAX` would probably not compile the code at all because the compiler can already figure out that the stack wouldn't be large enough for all these variables which are declared like `char buf[PATH_MAX]` etc.

So let's go with the simple diff I've recommended. Avoids introduction of yet another `sprintf` call.